ir_datasets/datasets/sara.py

```diff
@@ -0,0 +1,225 @@
+import ir_datasets
+from ir_datasets.formats import TsvDocs, TsvQueries, TrecQrels
+from ir_datasets.util import DownloadConfig, TarExtract, Cache
+from ir_datasets.datasets.base import Dataset, YamlDocumentation
+from ir_datasets.datasets.base import Dataset, FilteredQueries, FilteredQrels, YamlDocumentation
+from ir_datasets.formats import BaseDocs, TrecXmlQueries, TrecQrels, GenericQuery, GenericQrel
+from ir_datasets.indices import PickleLz4FullStore
+from typing import NamedTuple, Tuple
+import itertools
+import io
+from ir_datasets.util import GzipExtract, Cache, Lazy, TarExtractAll, RelativePath
+import os 
+import re
+import logging
+import glob
+import tarfile
+
+# Utility functions
+
+def filenames_from_cat(top,second, directory):
+    cats_globs = glob.glob(directory + '/*/*/*.cats')
+    matches = []
+    for g in cats_globs:
+        with open(g) as file:
+            for line in file:
+                line = line.split()[0]
+                line = line.split(",")
+                if int(line[0]) == top and int(line[1]) == second:
+                    dir = os.path.basename(g)
+                    filename = os.path.splitext(dir)[0]
+                    matches.append(filename)
+    return matches
+
+def print_email_from_filename(directory,filename):
```

Comment:
Some of these functions are not used. Can you remove the ones that are not needed?